### ISSUE TITLE
RocksdB backup sync

### ISSUE THREAD

**Issue by bharatviswa504** [facebook/rocksdb#4959]
![image](https://user-images.githubusercontent.com/8586345/52454740-c74a2e80-2b01-11e9-91c7-68903292d96f.png)

**Comment by bharatviswa504** [comment]
I have written a sample program and it is working on my machine, just want to know if it has any caveats or surprises by using this way?

**Comment by sagar0** [comment]
This is a tough question to answer without knowing a lot of additional details of your use-case and setup environment. For example:
- Are the two nodes exact replicas of each other?
- How is the traffic split between the two ... i.e. does your application use queuing service (say, like kafka) to write to one node (maybe primary/master), and offload the replication logic for slaves? If this is the case, the snapshots might be created at different logical points in time, and the data might not be the same in snapshot n on node 1 and snapshot n on node 2.
- eventual consistency / strong consistency? 

In short, it highly depends on how you implement your replication and backup/restore module. 

To answer your point 3: if you copy the **full** snapshot directory of node 1, yes, you can be assured that the data will be the same. But if you somehow manually decide to copy only the "delta" of node 1's snapshots (i.e. from n to n+m), I don't think the data will be the same. This is due to what i said above: the snapshots on nodes 1 and 2 could be applied at different points in time.

**Comment by bharatviswa504** [comment]
Thank You @sagar0  for info. 
We are using another API getUpdatesSince() together with the checkpoint to solve the issue.

**Comment by mars00772** [comment]
@bharatviswa504 how to use getUpdatesSince to get the certain key and value after one seq？
